src/components/form/widgets/date-picker.ts

```diff
@@ -53,15 +54,16 @@ export class DatePicker extends React.Component {
         }
     }
 
-    private handleChange(event: CustomEvent<boolean>) {
+    private handleChange(event: CustomEvent<Date>) {
         const props = this.props;
         event.stopPropagation();
 
         if (!props.onChange) {
             return;
         }
 
-        props.onChange(event.detail);
+        const dateString = moment(event.detail).format('YYYY-MM-DD');
```

Comment:
This seems to work on my machine, and I've got American dates because I have my browser set to English. So the hardcoded format here doesn't seem to cause any issues even though the format shown might be different. I'm not quite sure what happens to the value once submitted though? It's sent to the server I assume? Does it get converted to a full JSON date first?

Reply:
No the date submitted will be in the format specified here as a string. I tried `date.toISOString()` first, but the form would never be valid since the format is specified as `date` in the schema and must be a valid representation according to the "full-date" in RFC 3339 https://tools.ietf.org/html/rfc3339#section-5.6

Reply:
Maybe there is a better way than using Moment? :thinking:

Reply:
Hmm… flatpickr is already "validating" the value, I think, because if I enter something that can't be parsed as a date, and tab out of the field, the field is emptied. And if I enter something that can be interpreted as a date, the field is formatted to the "correct" format for that date.

So, I guess the only validation we really get to do on this is to check that it's not empty, because if it's not empty, it must be a valid date. At least I think so… 😄

Reply:
Hmm yeah maybe :thinking: Since the value from the event is a `Date` object, it must be a valid date I guess :smile:

Reply:
But we cannot put that object in the form value, since the schema wants a `string`

Reply:
> Maybe there is a better way than using Moment? 🤔

For transforming dates, I think we should stick to Moment 👍